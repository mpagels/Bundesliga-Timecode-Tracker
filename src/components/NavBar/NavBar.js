import React from 'react'
import styled from 'styled-components/macro'
import { getTimecodeTotalLengthFromSequenceCards } from '../../utils/Timecode'
import { Link, useLocation } from 'react-router-dom'

export default function NavBar({
  firstHalfTimeCode,
  secondHalfTimeCode,
  interviewTimeCode,
  countSpecials,
}) {
  const location = useLocation()
  return (
    <NavWrapper>
      <ul>
        <li>
          <Link to="/">
            <p>1ST</p>
            <span>
              {getTimecodeTotalLengthFromSequenceCards(firstHalfTimeCode)}
            </span>
          </Link>
          {location.pathname === '/' && <ActiveBar></ActiveBar>}
        </li>
        <li>
          <Link to="/2nd">
            <p>2ND</p>
            <span>
              {getTimecodeTotalLengthFromSequenceCards(secondHalfTimeCode)}
            </span>
            {location.pathname === '/2nd' && <ActiveBar></ActiveBar>}
          </Link>
        </li>
        <li>
          <Link to="/interview">
            <p>INTERVIEW</p>
            <span>
              {getTimecodeTotalLengthFromSequenceCards(interviewTimeCode)}
            </span>
            {location.pathname === '/interview' && <ActiveBar></ActiveBar>}
          </Link>
        </li>
        <li>
          <Link to="/special">
            <SpecialWrapper>
              {countSpecials > 0 && (
                <SpecialCouter>{countSpecials}</SpecialCouter>
              )}

              <p>SPECIALS</p>
            </SpecialWrapper>
            {location.pathname === '/special' && <ActiveBar></ActiveBar>}
          </Link>
        </li>
        <li>
          <Link to="/info">
            <p>INFO</p>
            {location.pathname === '/info' && <ActiveBar></ActiveBar>}
          </Link>
        </li>
      </ul>
    </NavWrapper>
  )
}

const NavWrapper = styled.nav`
  position: fixed;
  bottom: 0px;
  height: 60px;
  display: flex;
  justify-content: center;
  padding: 0;
  margin: 0;
  right: 0;
  left: 0;
  background: linear-gradient(
    180deg,
    rgba(var(--white-grey)) 0%,
    rgba(var(--white)) 100%
  );

  & ul {
    width: 100%;
    padding: 0;
    display: flex;
    align-items: center;
    justify-content: space-evenly;
  }
  & li {
    font-size: 0.6em;
    display: flex;
    flex-direction: column;
    align-items: center;
    list-style: none;
    padding: 10px;
    width: 20%;

    & a {
      text-decoration: none;
      display: flex;
      flex-direction: column;
      align-items: center;
    }

    & p {
      color: var(--font-blue);
      margin: 0;
      padding: 3px;
      font-weight: 700;
    }

    & span {
      color: var(--font-blue);
    }
  }
`

const ActiveBar = styled.div`
  position: absolute;
  bottom: 0;
  width: 20%;
  height: 3px;
  background-color: var(--font-blue);
  margin: 0;
  padding: 0;
`

const SpecialWrapper = styled.div`
  position: relative;
`

const SpecialCouter = styled.div`
  position: absolute;
  top: -15px;
  right: -15px;
  display: flex;
  justify-content: center;
  align-items: center;
  font-family: 'BaiJamjuree';
  font-size: 1em;
  color: white;
  background-color: var(--special-orange);
  border-radius: 50px;
  width: 18px;
  height: 18px;
`
