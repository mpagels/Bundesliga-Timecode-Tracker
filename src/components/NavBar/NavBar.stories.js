import React from 'react'
import NavBar from './NavBar'

export default {
  title: 'capstone/NavBar',
  component: NavBar,
}

const Template = (args) => <NavBar {...args} />

export const Empty = Template.bind({})
Empty.args = {
  firstHalfTimeCode: ['0000'],
  secondHalfTimeCode: ['0000'],
  interviewTimeCode: ['0000'],
  countSpecials: [],
}
export const Full = Template.bind({})
Full.args = {
  firstHalfTimeCode: [
    { timeCode: '2700', isActive: true },
    { timeCode: '2700', isActive: true },
    { timeCode: '2700', isActive: true },
  ],
  secondHalfTimeCode: [
    { timeCode: '2500', isActive: true },
    { timeCode: '2500', isActive: true },
    { timeCode: '2500', isActive: true },
  ],
  interviewTimeCode: [
    { timeCode: '2500', isActive: true },
    { timeCode: '2500', isActive: true },
    { timeCode: '2500', isActive: true },
  ],
  countSpecials: 3,
}
