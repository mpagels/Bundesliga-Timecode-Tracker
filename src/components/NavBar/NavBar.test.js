import React from 'react'
import NavBar from './NavBar'
import { render, cleanup } from '@testing-library/react'
import { Router } from 'react-router-dom'
import { createMemoryHistory } from 'history'

afterEach(cleanup)

it('renders correctly the NavBar', () => {
  const history = createMemoryHistory()
  const { asFragment, debug } = render(
    <Router history={history}>
      <NavBar
        firstHalfTimeCode={[
          { timeCode: '2700', isActive: true },
          { timeCode: '2700', isActive: true },
          { timeCode: '2700', isActive: true },
        ]}
        secondHalfTimeCode={[
          { timeCode: '2500', isActive: true },
          { timeCode: '2500', isActive: true },
          { timeCode: '2500', isActive: true },
        ]}
        interviewTimeCode={[
          { timeCode: '2500', isActive: true },
          { timeCode: '2500', isActive: true },
          { timeCode: '2500', isActive: true },
        ]}
        countSpecials={3}
      />
    </Router>
  )
  debug()
  expect(asFragment()).toMatchSnapshot()
})
