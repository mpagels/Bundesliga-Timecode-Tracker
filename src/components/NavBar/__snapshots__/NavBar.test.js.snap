// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly a small type header 1`] = `
<DocumentFragment>
  <nav
    class="NavBar__NavWrapper-jwnntt-0 LDDlK"
  >
    <ul>
      <li>
        <a
          href="/"
        >
          <p>
            1ST
          </p>
          <span>
            00:01:21:00
          </span>
        </a>
        <div
          class="NavBar__ActiveBar-jwnntt-1 fyAIhG"
        />
      </li>
      <li>
        <a
          href="/2nd"
        >
          <p>
            2ND
          </p>
          <span>
            00:01:15:00
          </span>
        </a>
      </li>
      <li>
        <a
          href="/interview"
        >
          <p>
            INTERVIEW
          </p>
          <span>
            00:01:15:00
          </span>
        </a>
      </li>
      <li>
        <a
          href="/special"
        >
          <div
            class="NavBar__SpecialWrapper-jwnntt-2 rvtLy"
          >
            <div
              class="NavBar__SpecialCouter-jwnntt-3 lcCauh"
            >
              3
            </div>
            <p>
              SPECIALS
            </p>
          </div>
        </a>
      </li>
      <li>
        <a
          href="/info"
        >
          <p>
            INFO
          </p>
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`renders correctly the NavBar 1`] = `
<DocumentFragment>
  <nav
    class="NavBar__NavWrapper-jwnntt-0 LDDlK"
  >
    <ul>
      <li>
        <a
          href="/"
        >
          <p>
            1ST
          </p>
          <span>
            00:01:21:00
          </span>
        </a>
        <div
          class="NavBar__ActiveBar-jwnntt-1 fyAIhG"
        />
      </li>
      <li>
        <a
          href="/2nd"
        >
          <p>
            2ND
          </p>
          <span>
            00:01:15:00
          </span>
        </a>
      </li>
      <li>
        <a
          href="/interview"
        >
          <p>
            INTERVIEW
          </p>
          <span>
            00:01:15:00
          </span>
        </a>
      </li>
      <li>
        <a
          href="/special"
        >
          <div
            class="NavBar__SpecialWrapper-jwnntt-2 rvtLy"
          >
            <div
              class="NavBar__SpecialCouter-jwnntt-3 lcCauh"
            >
              3
            </div>
            <p>
              SPECIALS
            </p>
          </div>
        </a>
      </li>
      <li>
        <a
          href="/info"
        >
          <p>
            INFO
          </p>
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
